### PR TITLE
imx6ull: usb phy_init implementation

### PR DESCRIPTION
**Pull request by badochov**
JIRA: RTOS-386

## Description
Implements usb phy on imx6ull.

## Motivation and Context
Usb phy relied on bootrom initialiation. Booting from flash doesn't provide this initialization.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: imx6ull

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: devices/usbc-cdc/usbphy-imx6ull.c [hunk 1]
```diff
@@ -50,19 +66,44 @@ void *phy_getBase(void)
...
+	while ((*CCM_ANALOG_PLL_USB0 & CCM_ANALOG_PLL_USB0_LOCK) == 0u) {
+	}
+	*CCM_ANALOG_PLL_USB0 &= ~(CCM_ANALOG_PLL_USB0_BYPASS);
+	*CCM_ANALOG_PLL_USB0 |= CCM_ANALOG_PLL_USB0_EN_USB_CLKS;
+}
+
 void phy_reset(void)
 {
-	/* TODO: reset phy controller */
+	USB0_PHY[USB_PHY_CTRL] |= USB_PHY_CTRL_SFTRST;
```
Comment (agkaminski) [inline devices/usbc-cdc/usbphy-imx6ull.c:87]: Our convention:
base address is kept in common struct, access to registers is performed via + operator:
```suggestion
	*(phyusb_common.base + USB_PHY_CTRL) |= USB_PHY_CTRL_SFTRST;
```

Comment (agkaminski) [reply devices/usbc-cdc/usbphy-imx6ull.c:87]: `phyusb_common.base` is to be initialized in `phy_init()`

**Review by agkaminski — DISMISSED** [review]

File: devices/usbc-cdc/usbphy-imx6ull.c (hunk 1 above)
Comment (agkaminski) [reply devices/usbc-cdc/usbphy-imx6ull.c:87]: Example: https://github.com/phoenix-rtos/plo/blob/f51d9744165a6914e631c83543282fa1c7b80348/devices/usbc-cdc/usbphy-imxrt117x.c#L101

**Review by gerard5 — COMMENTED** [review]
Looks good to me. Great job, USB now works as it does on other `plo` platforms. As for the style of access to the registers, you must agree with Alek.

File: devices/usbc-cdc/usbphy-imx6ull.c (hunk 1 above)
Comment (badochov) [reply devices/usbc-cdc/usbphy-imx6ull.c]: Is it good now?

Comment (agkaminski) [reply devices/usbc-cdc/usbphy-imx6ull.c]: `CCM_ANALOG_PLL_USB0` stayed in a different style

File: devices/usbc-cdc/usbphy-imx6ull.c [hunk 2]
```diff
@@ -50,19 +66,48 @@ void *phy_getBase(void)
...
+	return 75u;
 }
 
 
+static void phy_setClock(void)
+{
+	imx6ull_setDevClock(clk_usboh3, 0x03);
+
+	*CCM_ANALOG_PLL_USB0 |= CCM_ANALOG_PLL_USB0_ENABLE;
+	*CCM_ANALOG_PLL_USB0 |= CCM_ANALOG_PLL_USB0_POWER;
```
Comment (agkaminski) [inline devices/usbc-cdc/usbphy-imx6ull.c]:
```suggestion
	*CCM_ANALOG_PLL_USB0 |= CCM_ANALOG_PLL_USB0_ENABLE;
	hal_cpuDataMemoryBarrier();
	*CCM_ANALOG_PLL_USB0 |= CCM_ANALOG_PLL_USB0_POWER;
	hal_cpuDataMemoryBarrier();
```

**Review by agkaminski — COMMENTED** [review]
ARM core (even with use of volatiles in code) can delay and rearrange data bus operations. Because of this if it essential for some operations to be completed in the exact specified order, we need to use `dmb` instruction.

**Review by agkaminski — APPROVED** [review]
LGTM

File: devices/usbc-cdc/usbphy-imx6ull.c [hunk 3]
```diff
@@ -50,19 +66,56 @@ void *phy_getBase(void)
...
+	volatile u32 *usbphy = (u32 *)USB0_PHY;
+
+	*(usbphy + USB_PHY_CTRL) |= USB_PHY_CTRL_SFTRST;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_CTRL) &= ~USB_PHY_CTRL_CLKGATE;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_CTRL) &= ~USB_PHY_CTRL_SFTRST;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_PWD) = 0u;
+	hal_cpuDataMemoryBarrier();
```
Comment (gerard5) [inline devices/usbc-cdc/usbphy-imx6ull.c:101]: This is from the imx6ull, recent official SDK, they don't use barriers in between :)
2023-03-08-101017_1132x168_scrot

Comment (agkaminski) [reply devices/usbc-cdc/usbphy-imx6ull.c:101]: Well, it certainly won't hurt anything

**Pull request merged**